boefjes/boefjes/plugins/kat_dns/normalize.py

```diff
@@ -152,6 +155,14 @@ def register_record(record: DNSRecord) -> DNSRecord:
                         )
                     )
 
+                if isinstance(rr, CAA):
+                    recordvalue = str(rr)
+                    recordvalue = recordvalue.split(" ", 2)
+                    default_args["flags"] = min(max(0, int(recordvalue[0])), 255)
```

Comment:
If the flag has some strange value I don't think we should silently change that.

I am also not sure if DNS guarentees that we get 3 record values back or that the DNS record can also be malformed?

Reply:
The raw record can be malformed indeed, however dns python DNS class already does some formatting to fit it into its DNS answer class.
Ideally we should not do this, and instead don't parse the data from the DNS-server in the boefje, and just return it verbatim into the raw file. If then a record has 'other data' that doesn't fit the expected format we should yield a Finding informing the user of that record and its original malformed raw value string.